src/common.inc

```diff
@@ -54,22 +54,35 @@ COVERAGE = 0
 PKG_CONFIG ?= pkg-config
 HEADERS = $(wildcard *.h) $(wildcard *.hpp)
 
+ifeq ($(CLANG_FORMAT),)
 ifeq ($(shell command -v clang-format-3.8 > /dev/null && echo y || echo n), y)
-CLANG_FORMAT ?= clang-format-3.8
+export CLANG_FORMAT ?= clang-format-3.8
 else
-CLANG_FORMAT ?= clang-format
+export CLANG_FORMAT ?= clang-format
+endif
 endif
 
 GCOV_CFLAGS=-fprofile-arcs -ftest-coverage --coverage
 GCOV_LDFLAGS=-fprofile-arcs -ftest-coverage
 GCOV_LIBS=-lgcov
 
-osdep = $(1)_$(shell uname -s | tr "[:upper:]" "[:lower:]")$(2)
+ifeq ($(OS_KERNEL_NAME),)
+export OS_KERNEL_NAME := $(shell uname -s)
```

Comment:
This will not work as expected. "?=" is like "=", in the sense that it does not immediately evaluate the expression on the right. We would need an operator that combines behaviors of ?= and :=. Unfortunately there is no such thing.